ecs/pod.go

```diff
@@ -162,17 +163,23 @@ func (p *BasicECSPod) Delete(ctx context.Context) error {
 		}
 	}
 
-	for _, secret := range p.resources.Secrets {
-		if utility.FromBoolPtr(secret.Owned) {
-			catcher.Wrap(p.vault.DeleteSecret(ctx, utility.FromStringPtr(secret.Name)), "deleting secret")
+	for _, c := range p.resources.Containers {
+		for _, s := range c.Secrets {
+			if utility.FromBoolPtr(s.Owned) {
+				id := utility.FromStringPtr(s.Name)
+				catcher.Wrapf(p.vault.DeleteSecret(ctx, id), "deleting secret '%s' for container '%s'", id, utility.FromStringPtr(c.Name))
+			}
 		}
 	}
 
 	if catcher.HasErrors() {
 		return catcher.Resolve()
 	}
 
-	p.status = cocoa.StatusDeleted
+	p.statusInfo.Status = cocoa.StatusDeleted
```

Comment:
The individual containers can be in different states. For example, you could have one stopped container, one running container, and one provisioning container but the entire task status would be running. This is mostly relevant when `GetLatestStatus` is implemented, which will reflect any differences in container statuses.

Reply:
If we only let you start/stop/delete the entire pod _as a group_ how would one container end up with a different state than the rest?

Reply:
You can easily simulate this. Start two pods, one which sleeps for 1 second and one which sleeps for 1 minute. The first pod will show as stopped almost immediately but the other one's status will be running for the entire time. If you assume that some containers could be transient and others could be persistent processes like services, the containers have completely independent lifecycles.

Just because we provide manual options to stop/delete doesn't prevent the individual containers from transitioning state.